developer/src/kmc-package/src/compiler/kmp-compiler.ts

```diff
@@ -82,32 +81,32 @@ export class KmpCompiler {
     // Fill in additional fields
     //
 
-    if(kps.options) {
-      kmp.options.executeProgram = kps.options?.executeProgram || undefined;
-      kmp.options.graphicFile = kps.options.graphicFile || undefined;
-      kmp.options.msiFilename = kps.options.msiFileName || undefined;
-      kmp.options.msiOptions = kps.options.msiOptions || undefined;
-      kmp.options.readmeFile = kps.options.readMeFile || undefined;
-      kmp.options.licenseFile = kps.options.licenseFile || undefined;
-      kmp.options.welcomeFile = kps.options.welcomeFile || undefined;
+    if(kps.Options) {
+      kmp.options.executeProgram = kps.Options?.ExecuteProgram || undefined;
+      kmp.options.graphicFile = kps.Options.GraphicFile || undefined;
+      kmp.options.msiFilename = kps.Options.MsiFileName || undefined;
+      kmp.options.msiOptions = kps.Options.MsiOptions || undefined;
+      kmp.options.readmeFile = kps.Options.ReadMeFile || undefined;
+      kmp.options.licenseFile = kps.Options.LicenseFile || undefined;
+      kmp.options.welcomeFile = kps.Options.WelcomeFile || undefined;
```

Comment:
Can't this be simplified since we already checked for `kps.Options`in L84?

```suggestion
      kmp.options.executeProgram = kps.Options.ExecuteProgram;
      kmp.options.graphicFile = kps.Options.GraphicFile;
      kmp.options.msiFilename = kps.Options.MsiFileName;
      kmp.options.msiOptions = kps.Options.MsiOptions;
      kmp.options.readmeFile = kps.Options.ReadMeFile;
      kmp.options.licenseFile = kps.Options.LicenseFile;
      kmp.options.welcomeFile = kps.Options.WelcomeFile;
```

Reply:
The `|| undefined` converts falsy values (`0`, `""`, `null`, `undefined`) to `undefined`, so that's deliberate and needs to stay. The `?` on `kps.Options?.ExecuteProgram` is not needed.

Reply:
```suggestion
      kmp.options.executeProgram = kps.Options.ExecuteProgram || undefined;
      kmp.options.graphicFile = kps.Options.GraphicFile || undefined;
      kmp.options.msiFilename = kps.Options.MsiFileName || undefined;
      kmp.options.msiOptions = kps.Options.MsiOptions || undefined;
      kmp.options.readmeFile = kps.Options.ReadMeFile || undefined;
      kmp.options.licenseFile = kps.Options.LicenseFile || undefined;
      kmp.options.welcomeFile = kps.Options.WelcomeFile || undefined;
```